Source/WebCore/page/PointerCaptureController.cpp

```diff
@@ -384,16 +385,24 @@ void PointerCaptureController::pointerEventWillBeDispatched(const PointerEvent&
     if (!isPointerdown && !isPointerup)
         return;
 
+    auto handlePopoverLightDismiss = [&]() {
+        if (auto* element = dynamicDowncast<HTMLElement>(target))
+            element->document().handlePopoverLightDismiss(event, *element);
+    };
+
     auto pointerId = event.pointerId();
 
     if (event.pointerType() != touchPointerEventType()) {
         if (RefPtr capturingData = m_activePointerIdsToCapturingData.get(pointerId))
             capturingData->pointerIsPressed = isPointerdown;
+        handlePopoverLightDismiss();       
         return;
     }
 
-    if (!isPointerdown)
+    if (!isPointerdown) {
+        handlePopoverLightDismiss();       
         return;
+    }
```

Comment:
I think we can change the logic a bit to avoid repeating the call to handlePopoverLightDismiss E.g.

```
if (event.pointerType() != touchPointerEventType()) {
   if (RefPtr capturingData = m_activePointerIdsToCapturingData.get(pointerId))
      capturingData->pointerIsPressed = isPointerdown;
} else if (isPointerDown) {
  auto capturingData = ...;
  [...]
}
element->document().handlePopoverLightDismiss(event, *element);